macOS Big Sur is 11.0, Darwin 20, after a long line of 10.X releases.
macOS Monterey is 12.0.
--- a/m4/libtool.m4
+++ b/m4/libtool.m4
@@ -1076,7 +1076,7 @@
 	  _lt_dar_allow_undefined='$wl-undefined ${wl}dynamic_lookup' ;;
 	10.[[012]][[,.]]*)
 	  _lt_dar_allow_undefined='$wl-flat_namespace $wl-undefined ${wl}suppress' ;;
-	10.*)
+	10.*|11.*|12.*)
 	  _lt_dar_allow_undefined='$wl-undefined ${wl}dynamic_lookup' ;;
       esac
     ;;
